sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetPartitionReaderFactory.scala

```diff
@@ -79,6 +83,25 @@ case class ParquetPartitionReaderFactory(
   private val pushDownInFilterThreshold = sqlConf.parquetFilterPushDownInFilterThreshold
   private val datetimeRebaseModeInRead = parquetOptions.datetimeRebaseModeInRead
   private val int96RebaseModeInRead = parquetOptions.int96RebaseModeInRead
+  private val columnBatchSize = sqlConf.columnBatchSize
+
+  private def getFooter(file: PartitionedFile): ParquetMetadata = {
```

Comment:
I think we should consider the start and length of the `file` and only do aggregation on those filtered row group metadata?